fuel-txpool/src/txpool.rs

```diff
@@ -319,36 +307,31 @@ impl TxPool {
     /// When block is updated we need to receive all spend outputs and remove them from txpool.
     pub async fn block_update(
         txpool: &RwLock<Self>,
+        tx_status_sender: &TxStatusChange,
         block: Arc<FuelBlock>,
         // spend_outputs: [Input], added_outputs: [AddedOutputs]
     ) {
         let mut guard = txpool.write().await;
         // TODO https://github.com/FuelLabs/fuel-core/issues/465
 
         for tx in block.transactions() {
+            tx_status_sender.send_complete(tx.id());
```

Comment:
No because there is nothing different we can do based on those states and it creates more complexity trying to handle them in the subscription without adding any value.

Reply:
Ohh, wait, you use it for the subscription too, but only like a trigger to make a request into DB/TxPool. @freesig Do we want to do that when we can broadcast the status itself and reduce the number of queries in DB?

Reply:
We can return statuses inside of `ExecutionResult` ;) And in that case, you will have them and don't need access to DB.

Reply:
But what about the submitted status?

Reply:
What with it? `TxPool` sends this event

Reply:
Ok I can see how this could be done. Will make an issue for this as tech debt

Reply:
Cool<3